### PR TITLE
[slack] add support to create user groups

### PR DESCRIPTION
**Pull request by maorfr**
https://coreos.slack.com/archives/C03KQ9M1P6C/p1662138005941939

ref: https://slack.com/help/articles/212906697-Create-a-user-group

### CODE REVIEW

**Comment by maorfr** [comment]
doc update: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/46716

**Review by patjlm — APPROVED** [review]
LGTM

**Review by geoberle — APPROVED** [review]

**Comment by maorfr** [comment]
it's not working as expected. redoing this.

File: reconcile/slack_usergroups.py [hunk 1]
```diff
@@ -73,10 +77,13 @@ def get_current_state(slack_map):
...
     for workspace, spec in slack_map.items():
-        slack = spec["slack"]
+        slack: SlackApi = spec["slack"]
         managed_usergroups = spec["managed_usergroups"]
         for ug in managed_usergroups:
-            users, channels, description = slack.describe_usergroup(ug)
+            try:
+                users, channels, description = slack.describe_usergroup(ug)
+            except UsergroupNotFoundException:
+                continue
```
Comment (maorfr) [inline reconcile/slack_usergroups.py:86]: when a usergroup is not found, it is not added to the current state

File: reconcile/slack_usergroups.py [hunk 2]
```diff
@@ -298,6 +305,34 @@ def get_desired_state(slack_map, pagerduty_map):
...
+    dry_run: bool = True,
+):
+    """
+    Create Slack usergroups.
+    """
+
+    workspace = desired_ug_state["workspace"]
+    usergroup = desired_ug_state["usergroup"]
+
+    if current_ug_state:
```
Comment (maorfr) [inline reconcile/slack_usergroups.py:321]: when current usergroup state exists, the usergroup exists

File: reconcile/slack_usergroups.py [hunk 3]
```diff
@@ -298,6 +305,34 @@ def get_desired_state(slack_map, pagerduty_map):
...
+        logging.debug(
+            f"[{workspace}] Usergroup exists and will not be created {usergroup}"
+        )
+        return
+
+    logging.info(["create_usergroup", workspace, usergroup])
+    if not dry_run:
+        try:
+            usergroup_id = slack_client.create_usergroup(usergroup)
+            desired_ug_state["usergroup_id"] = usergroup_id
```
Comment (maorfr) [inline reconcile/slack_usergroups.py:331]: once the usergroup is created, update the usergroup id in the desired state for following actions (adding users, etc)

**Comment by maorfr** [comment]
now it's ready :pray:

**Review by geoberle — APPROVED** [review]

**Pull request merged**